### DISCUSSION TITLE
aptos Prover Question

### DISCUSSION THREAD

**Discussion by yue1823** [aptos-labs/aptos-developer-discussions#352] [Questions]
### Discord user ID

._yue_.

### Describe your question in detail.

when i trying to use `aptos move prove` ,show this error 
![image](https://github.com/user-attachments/assets/ec156bbc-9001-4d4b-85f3-f09b5b2d2d68)
Therefore , I use this command `source ~/.profile`
Then I get another error
<img width="1429" alt="image" src="https://github.com/user-attachments/assets/bd027fc6-5e2f-461a-a12b-41605e684e57">

Anyone can help me to solve this error 

### Which operating system are you using?

macOS Macbook Pro m1 Arm

### Which SDK or tool are you using? (if any)

Aptos CLI

### Describe your environment or tooling in detail

aptos move prove on idea

**Comment by WGB5445** [comment]
Can you mark your system, version and computer architecture? This may be more helpful in solving the problem.

Also, have you tried restarting the terminal or computer? At first glance, it seems that it may be due to an environment variable issue.

**Reply by yue1823** [reply]
i am using macbook m1 arm , restart seems not help in this situation ( I try it )

**Comment by MoonShiesty** [comment]
dont think this is related to your environment


can you post your source-code/specs you're proving?

ive seen this error before can't remember how to reproduce. dont remember the details but i think it happens when you call non-spec functions outside of your module in a spec. i imagine if you start to comment out the specs and re-run the prover itll work again at some point

**Reply by yue1823** [reply]
I am not set any  spec in /source ,actually I don't know what's the spec you are refer to

**Reply by yue1823** [reply]
<img width="307" alt="image" src="https://github.com/user-attachments/assets/2977ac3e-d641-4942-8fdc-27db72164c14">

**Reply by yue1823** [reply]
<img width="1009" alt="截圖 2024-08-01 上午1 00 37" src="https://github.com/user-attachments/assets/268bf5dd-2496-4749-ae72-699f05fbf9f5">

the error is  for this one
` #[lint::allow_unsafe_randomness]`
but i dont think this would be the reason of prove error

**Reply by MoonShiesty** [reply]
im just starting to learn the prover, and have no idea how attributes are processed. i imagine aptos needs to add something to randomness.spec.move

i also see that aptos added this attribute to aptos-core at the end of june, id double check you're using the latest aptos-core and cli

for now you should be able to turn off verification for that function instead of at the module scope, and see if you can prove the rest of your project

try putting this at the top of the function that uses `allow_unsafe_randomness`, and turn back on verification at the module level:
`pragma verify = false`

**Comment by yue1823** [comment]
if i add 
```  
spec module {
        pragma verify = false; // TODO: see issue <url>
    }
```
    
  ``` move
  [INFO] transforming bytecode
[INFO] generating verification conditions
[INFO] 16 verification conditions
[INFO] running solver
[INFO] 1.625s build, 3.977s trafo, 0.048s gen, 6.090s verify, total 11.741s
{
  "Result": "Success"
}

 ```

**Comment by 0xbrw** [comment]
@yue1823 can you share your entire code file and the commands you ran so I can try to reproduce the error on my side?

**Reply by yue1823** [reply]
[game.zip](https://github.com/user-attachments/files/16448108/game.zip)
```
[package]
name = "chips"
version = "1.0.0"
authors = []

[addresses]
dapp="0x2"
admin="0x555"
admin1="0xeee"
admin2="0xa51"
[dev-addresses]

[dependencies.AptosFramework]
git = "https://github.com/aptos-labs/aptos-core.git"
rev = "mainnet"
subdir = "aptos-move/framework/aptos-framework"

[dependencies.AptosTokenObjects]
git = "https://github.com/aptos-labs/aptos-core.git"
#git = "https://gitee.com/WGB5445/aptos-core.git"
rev = "mainnet"
subdir = "aptos-move/framework/aptos-token-objects"

[dev-dependencies]

```

please dont share to others

**Reply by 0xbrw** [reply]
I was able to run the aptos prover against your code. 
```
[INFO] preparing module 0x1::game
[INFO] preparing module 0x2::helper
[INFO] transforming bytecode
[INFO] generating verification conditions
[INFO] 16 verification conditions
[INFO] running solver
[INFO] 11.909s build, 21.845s trafo, 0.296s gen, 38.918s verify, total 72.969s
{
  "Result": "Success"
}
```

The problem occurs if you have installation issues with the move prover. A good way to test is to run the prover in the example here: https://aptos.dev/en/build/cli/setup-cli/install-move-prover If that works, you can copy over your project to the `/move-examples` sub-directory and run the prover from there

**Reply by yue1823** [reply]
You run success results since I set
‘
spec module {
        pragma verify = false; // TODO: see issue <url>
    }
‘
on lowest of my code, could you try to close it and run again?

**Comment by wrwg** [comment]
Please follow the instructions on how to install the Move Prover auxiliary tools [here](https://aptos.dev/en/build/cli/setup-cli/install-move-prover). If the variable `BOGGIE_EXE` is not set, it indicates that the tools are not correctly configured. Unfortunately, for this you need to fork the aptos-core repo. (However, you only need the source to execute a shell script in there, no build of anything needed inside there.)

We are working on making this experience better.

**Comment by wrwg** [comment]
@yue1823 @bwdotcom there is 2nd issue apart of setting up the Move Prover correctly, thanks for brining it up! I isolated the cause and opened a [bug for it](https://github.com/aptos-labs/aptos-core/issues/14205). Fix should be forthcoming, you can follow along at the bug.

**Reply by yue1823** [reply]
i also find this error from my code when i try to use `batch_transfer_coins<CoinA>()` this function ,seems is a bug  like yours report

**Comment by StrawberryChocolateFudge** [comment]
I have the same issue
Using linux I installed boogie with `sudo apt install boogie`, installing with dotnet was not working out at all
so I have boogie at /usr/bin/boogie 
Boogie program verifier version 2.4.1.10503, Copyright (c) 2003-2014, Microsoft.

Now my BOOGIE_EXE is set to /usr/bin/boogie

The error from the prover is :

```
{
  "Error": "Move Prover failed: cannot extract version from `/usr/bin/boogie`"
}

```

**Comment by wrwg** [comment]
Please do not install boogie manually. Use `aptos update prover-dependencies` instead.
